src/wasm-linker.cpp

```diff
@@ -208,16 +214,19 @@ bool Linker::linkObject(S2WasmBuilder& builder) {
   // Allow duplicate aliases only if they refer to the same name. For now we
   // do not expect aliases in compiler-rt files.
   // TODO: figure out what the semantics of merging aliases should be.
-  for (const auto& alias : newSymbols->aliasedFunctions) {
-    if (out.symbolInfo.aliasedFunctions.count(alias.first) &&
-        out.symbolInfo.aliasedFunctions[alias.first] != alias.second) {
+  for (const auto& alias : newSymbols->aliasedSymbols) {
+    if (out.symbolInfo.aliasedSymbols.count(alias.first) &&
+      out.symbolInfo.aliasedSymbols.at(alias.first).symbol != alias.second.symbol) {
```

Comment:
We should probably also emit an error if the symbols match but the types do not.